src/wallet.ts

```diff
@@ -500,7 +500,7 @@ WalletMiddlewareOptions): JsonRpcMiddleware<any, Block> {
  */
 function validateVerifyingContract(data: string) {
   const { domain: { verifyingContract } = {} } = parseTypedMessage(data);
-  if (verifyingContract && !isValidHexAddress(verifyingContract)) {
+  if (verifyingContract && !(isValidHexAddress(verifyingContract) || verifyingContract == 'cosmos')) {
```

Comment:
I think it will be better to support any bech32 format for cosmos chain.

I addition can you also plz add comment detailing why we have done this change.

Reply:
thx.
currently we want to solve the hard coded "cosmos" issue, no need for generic bech32 support.
fixed and opened PR to the origin